### PR TITLE
chore: `provisionSmartWallet` flake

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
## Description
- motivated after observing `cannot read data of published.wallet.${addr}.current: fetch failed` in [10638](https://github.com/Agoric/agoric-sdk/actions/runs/12313012295/job/34369641775?pr=10638)

### Security Considerations
None

### Scaling Considerations
None

### Documentation Considerations
None

### Testing Considerations
None

### Upgrade Considerations
None

### CODE REVIEW

File: multichain-testing/test/fast-usdc/fast-usdc.test.ts [hunk 1]
```diff
@@ -50,17 +50,13 @@ test.before(async t => {
...
+    keys(oracleMnemonics).map(n =>
+      provisionSmartWallet(wallets[n], {
+        BLD: 100n,
+      }),
+    ),
   );
-  // execute sequentially, to avoid "published.wallet.${addr}.current: fetch failed"
-  const oracleWds: WalletDriver[] = [];
-  for (const p of oracleWdPs) {
-    const wd = await p;
```
Comment (dckc) [inline multichain-testing/test/fast-usdc/fast-usdc.test.ts:61]: yeah... that never did execute sequentially; that is: it didn't wait for one to finish before starting the next.

File: multichain-testing/test/support.ts [hunk 2]
```diff
@@ -79,13 +79,13 @@ export const commonSetup = async (t: ExecutionContext) => {
     console.error('setupRegistry failed', e);
     throw e;
   }
-  const tools = await makeAgdTools(t.log, childProcess);
-  const keyring = await makeKeyring(tools);
-  const deployBuilder = makeDeployBuilder(tools, fse.readJSON, execa);
   const retryUntilCondition = makeRetryUntilCondition({
     log: t.log,
     setTimeout: globalThis.setTimeout,
```
Comment (dckc) [inline multichain-testing/test/support.ts]: only top level scripts (incl. test scripts) should be grabbing ambient authority like this use of `globalThis.setTimeout`.

pre-dates this PR, though, so I'll add it to the list:
 - #2160

File: multichain-testing/tools/sleep.ts [hunk 3]
```diff
@@ -28,11 +28,11 @@ const retryUntilCondition = async <T>(
   {
     maxRetries = 6,
     retryIntervalMs = 3500,
-    log = () => {},
+    log = console.log,
```
Comment (turadg) [inline multichain-testing/tools/sleep.ts:31]: personally I find  this log output noisy. Good to change `console.log` below to this `log`, but consider leaving the noop default for silence

**Review by turadg — APPROVED** [review]

File: multichain-testing/test/support.ts [hunk 4]
```diff
@@ -79,13 +79,13 @@ export const commonSetup = async (t: ExecutionContext) => {
...
   });
+  const tools = await makeAgdTools(t.log, childProcess, retryUntilCondition);
```
Comment (dckc) [inline multichain-testing/test/support.ts]: though... now `tools` and everything derived from it carries the ambient authority along.

`childProcess` was already ambient authority though.

File: multichain-testing/tools/e2e-tools.js [hunk 5]
```diff
@@ -187,7 +192,12 @@ export const provisionSmartWallet = async (
     { chainId, from: address, yes: true },
   );
 
-  const info = await q.queryData(`published.wallet.${address}.current`);
+  const info = await retryUntilCondition(
+    () => q.queryData(`published.wallet.${address}.current`),
+    result => !!result,
+    `wallet in vstorage ${address}`,
+    { log: () => {} }, // suppress logs as this is already noisy
```
Comment (dckc) [inline multichain-testing/tools/e2e-tools.js:200]: yeah; we do too much logging without giving the caller the ability to turn it off.

File: multichain-testing/tools/sleep.ts (hunk 3 above)
Comment (0xpatrickdev) [reply multichain-testing/tools/sleep.ts:31]: This (+ `console.log -> log`) would impact a lot of existing tests, so I chose to preserve the existing behavior. I'll see how this looks in local runs and will look to land it in the future:
```diff
+    log = () => {},
-    log = console.log,
```

File: multichain-testing/tools/e2e-tools.js [hunk 6]
```diff
@@ -438,6 +449,7 @@ export const makeE2ETools = async (
     setTimeout,
     rpcAddress = 'http://localhost:26657',
     apiAddress = 'http://localhost:1317',
+    retryUntilCondition,
```
Comment (dckc) [inline multichain-testing/tools/e2e-tools.js]: `setTimeout` is already in the list... I suppose somebody might want the `log` in `retryUntilCondition` to be different from the `log` passed to `makeE2ETools`, but in the common case, we can fill it in for them:

```suggestion
    setTimeout,
    rpcAddress = 'http://localhost:26657',
    apiAddress = 'http://localhost:1317',
    retryUntilCondition = makeRetryUntilCondition({ log, setTimeout }),
```

**Review by dckc — APPROVED** [review]
LGTM

I think it's worthwhile making `retryUntilCondition` optional on `makeE2ETools`. But it's not critical.

File: multichain-testing/test/fast-usdc/fast-usdc.test.ts (hunk 1 above)
Comment (0xpatrickdev) [reply multichain-testing/test/fast-usdc/fast-usdc.test.ts:61]: Oh right, they all start together

File: multichain-testing/tools/e2e-tools.js (hunk 6 above)
Comment (0xpatrickdev) [reply multichain-testing/tools/e2e-tools.js]: Nice, good thinking

File: multichain-testing/test/support.ts (hunk 2 above)
Comment (turadg) [reply multichain-testing/test/support.ts]: I thought we agreed modules that are used only in testing can have ambient authority? 

(btw, I bristle at the subjectivity and emotional valence of that issue title. `s/Distressing //`)

File: multichain-testing/tools/sleep.ts (hunk 3 above)
Comment (turadg) [reply multichain-testing/tools/sleep.ts:31]: s/impact/denoise 😉

**Pull request merged**